lib/coinbase/wallet.rb

```diff
@@ -99,6 +99,24 @@ def network_id
       Coinbase.to_sym(@model.network_id)
     end
 
+    # Sets the seed of the Wallet. This seed is used to derive keys and sign transactions.
+    # @param seed [String] The seed to set. Expects a 32-byte hexadecimal with no 0x prefix.
+    def seed=(seed)
+      raise ArgumentError, 'Seed must be 32 bytes' if seed.length != 64
+      raise 'Seed is already set' unless @master.nil?
+      raise 'Cannot set seed for Wallet with non-zero private key index' if @private_key_index.positive?
+
+      @master = MoneyTree::Master.new(seed_hex: seed)
+
+      @addresses.each do
+        key = derive_key
+        address = address(key.address.to_s)
```

Comment:
Can we avoid method and variable shadowing here?

I feel like we should probably just get rid of the `address` function that takes an argument since it's a bit confusing

Reply:
Done